graphql-dgs-codegen-shared-core/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/BaseSubProjectionNode.kt

```diff
@@ -18,19 +18,60 @@ package com.netflix.graphql.dgs.client.codegen
 
 import java.util.*
 
-abstract class BaseSubProjectionNode<T, R>(
+abstract class BaseSubProjectionNode<S, T, R>(
     val parent: T,
     val root: R,
     schemaType: Optional<String> = Optional.empty()
 ) : BaseProjectionNode(schemaType) {
 
     constructor(parent: T, root: R) : this(parent, root, schemaType = Optional.empty())
-
     fun parent(): T {
         return parent
     }
 
     fun root(): R {
         return root
     }
+
+    fun alias(alias: String, assigner: (node: S) -> Any): S {
+        // Avoid alias against existing field name
+        if (this.javaClass.kotlin.members.any { it.name.equals(alias, ignoreCase = true) }) {
+            throw AssertionError("Tried to specify alias $alias which already exists as field.")
+        }
+
+        // Stash our current set of fields
+        val currentFields = fields.toMap()
+        val currentInputArguments = inputArguments.toMap()
+        fields.clear()
+        inputArguments.clear()
+
+        // Track the aliased field
+        @Suppress("UNCHECKED_CAST")
+        assigner(this as S)
```

Comment:
@srinivasankavitha I had a tough time getting this to become type safe. The purpose of this casting is for `assigner` to receive the the current projection node, casted as the first type variable, so that developers don't need to cast within `assigner` calls. 

It all worked as far as building and tests passing when I added this `@Suprress` (more validation needed). I suspect this is a language quirk but wanted to flag this for attention.

Reply:
That makes sense, thanks for the explanation.